pkg/api/doc.go

```diff
@@ -0,0 +1,82 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```

Comment:
This change seems unrelated to the scope of the PR. In the  future it will be good to have these in their own PRs

Reply:
I added it so that `WithRegion()` is documented in this PR directly rather than have to wait for another PR cycle.